Apps/frontend/src/components/WaveControls.svelte

```diff
@@ -7,10 +7,48 @@
   import { timeSweep } from "../stores.js";
   import TimeSweepSlider from "./TimeSweepSlider.svelte";
   import ThicknessSwitch from "./ThicknessSwitch.svelte";
+  import DistributeOffsetButton from "./DistributeOffsetButton.svelte";
+  import startStopLine from "./Waves.svelte";
   export let waveElement;
   export let isEnabled;
   export let indicatorElement;
   let btnOnOff;
+
+  let activeChannels = [];
+  let activeChannelCounter = NUM_CHANNELS;
+
+  let distributed = false;
+
+  const setActiveChannels = () => {
+    for (let i = 0; i < NUM_CHANNELS; i++) {
+      activeChannels[i] = false;
+    }
+  };
+
+  setActiveChannels();
+
+  const countActiveChannels = () => {
+    let count = 0;
+    for (let i = 0; i < NUM_CHANNELS; i++) {
+      if (activeChannels[i]) count++;
+    }
+    activeChannelCounter = count;
+  };
+
+  const distributeChannels = () => {
+    let offset = 2 / (activeChannelCounter + 1);
+    // loop over all channels and set offset
+    let offsetY = 1 - offset;
+    for (let index = 0; index < NUM_CHANNELS; index++) {
+      console.log(offsetY);
+      if (activeChannels[index]) {
+        waveElement.updateChannelOffsetY(index, offsetY);
+        indicatorElement.updateChannelOffsetY(index, offsetY);
+        document.getElementById(`offsetSlider-${index}`).value = offsetY;
```

Comment:
Please don't use `document.getElementById`, introduce a function inside the slider component to set the value there.